### ISSUE TITLE
addr2info() not available on OpenBSD

### ISSUE THREAD

**Issue by kusalananda** [att/ast#1356]
**Description of problem:**

OpenBSD (-current, on amd64) does not have `addr2info()`, used by `src/lib/libast/tests/misc/debug.c`, and I don't know a package that supplies it (it's not tested for by the Meson configuration, AFAIK).

**Ksh version:**

Since commit b90e7e377714dcc20f404f4b02e8ab558c8beab7

**How reproducible:**

Always.

**Steps to reproduce:**
1. Check out b90e7e377714dcc20f404f4b02e8ab558c8beab7 and compile.

**Actual results:**

```
$ ninja -v -C build
ninja: Entering directory `build'
[1/8] ccache cc  -o src/lib/libast/tests/misc/debug 'src/lib/libast/tests/misc/258ceee@@debug@exe/debug.c.o' -Wl,--as-needed -Wl,--export-dynamic -Wl,--start-group src/lib/libast/libast.a src/l
ib/libast/libenv.a -lm /usr/local/lib/libiconv.so.7.0 -Wl,--end-group -Wl,-z,origin '-Wl,-rpath,$ORIGIN/../..' -Wl,-rpath-link,/home/kk/local/build/ast/build/src/lib/libast
FAILED: src/lib/libast/tests/misc/debug
ccache cc  -o src/lib/libast/tests/misc/debug 'src/lib/libast/tests/misc/258ceee@@debug@exe/debug.c.o' -Wl,--as-needed -Wl,--export-dynamic -Wl,--start-group src/lib/libast/libast.a src/lib/lib
ast/libenv.a -lm /usr/local/lib/libiconv.so.7.0 -Wl,--end-group -Wl,-z,origin '-Wl,-rpath,$ORIGIN/../..' -Wl,-rpath-link,/home/kk/local/build/ast/build/src/lib/libast
ld: error: undefined symbol: addr2info
>>> referenced by debug.c:25 (../src/lib/libast/tests/misc/debug.c:25)
>>>               src/lib/libast/tests/misc/258ceee@@debug@exe/debug.c.o:(test_addr2info)
cc: error: linker command failed with exit code 1 (use -v to see invocation)
[2/8] /usr/local/bin/meson --internal vcstagger ../src/cmd/ksh93/include/version.h.in src/cmd/ksh93/include/version.h 2020.0.0-alpha1 /home/kk/local/build/ast/src/cmd/ksh93/include @VCS_TAG@ '(
.*)' /usr/local/libexec/git/git describe --always --dirty --tags
ninja: build stopped: subcommand failed.
```

**Expected results:**

Sucessful build.

**Additional info:**

```
$ uname -a
OpenBSD eeyore.my.domain 6.5 GENERIC#121 amd64
```

**Comment by kusalananda** [comment]
Duh, I'm a doofus. The function is implemented in `src/lib/libast/misc/debug.c`, but for some reason it's not getting compiled on my system.

I'm wondering if it's the `execinfo` library that is causing issues (again).

**Comment by krader1961** [comment]
@kusalananda, My apologies. This is because I recently added unit tests for `addr2info()` and the other functions in the *src/lib/libast/misc/debug.c* module. However, I didn't predicate those tests on whether preprocessor symbol `_hdr_execinfo` is true like the *debug.c* module does. This also highlights that there isn't an equivalent for `addr2info()` when `_hdr_execinfo` is false and there should be.

**Comment by krader1961** [comment]
@kusalananda I've pushed a fix for this issue. I simulated *execinfo.h* not being available to confirm commit #d27dba60 seems to fix this issue. If commit #d27dba60 doesn't fix this issue please reopen it.

**Comment by kusalananda** [comment]
Thanks! The code currently on the `master` branch compiles with the native Clang (8.0.0) compiler as well as with GCC (8.3.0) from ports on OpenBSD.
